### PR TITLE
remove DaemonSet recommendation; add warning

### PR DESCRIPTION
**Pull request by taroface**
Fixes [DOC-1623](https://cockroachlabs.atlassian.net/browse/DOC-1623).

- Added DaemonSet warning to Kubernetes Performance doc
- Removed DaemonSet recommendations in all docs versions
- Some opportunistic updates to Kubernetes Performance doc (which remains in need of a broader update)

### CODE REVIEW

File: v21.2/kubernetes-performance.md [hunk 1]
```diff
@@ -5,20 +5,20 @@ toc: true
...
+A number of independent factors affect performance when running CockroachDB on Kubernetes. Most are easiest to change before you create your CockroachDB cluster. If you need to modify a CockroachDB cluster that is already running on Kubernetes, extra care and testing is strongly recommended.
 
 In a number of the sections below, we have shown how to modify excerpts from our provided Kubernetes configuration YAML files. You can find the most up-to-date versions of these files on Github, [one for running CockroachDB in secure mode](https://github.com/cockroachdb/cockroach/blob/master/cloud/kubernetes/cockroachdb-statefulset-secure.yaml) and one for [running CockroachDB in insecure mode](https://github.com/cockroachdb/cockroach/blob/master/cloud/kubernetes/cockroachdb-statefulset.yaml).
```
Comment (MichaelTrestman) [inline v21.2/kubernetes-performance.md:23]:
```suggestion
The following sections show how to modify excerpts from our provided Kubernetes configuration YAML files. You can find the most up-to-date versions of these files on Github, [one for running CockroachDB in secure mode](https://github.com/cockroachdb/cockroach/blob/master/cloud/kubernetes/cockroachdb-statefulset-secure.yaml) and one for [running CockroachDB in insecure mode](https://github.com/cockroachdb/cockroach/blob/master/cloud/kubernetes/cockroachdb-statefulset.yaml).
```

File: v21.2/kubernetes-performance.md [hunk 2]
```diff
@@ -158,15 +158,13 @@ Since [GCE disk IOPS scale linearly with disk size](https://cloud.google.com/com
...
-
-There is also the option of using local disks if you do not run CockroachDB in a `StatefulSet`, but instead use a `DaemonSet`. For more details on what this entails, see the section on [Running in a DaemonSet](#running-in-a-daemonset).
+Up to this point, we have been assuming that you will be using auto-provisioned remotely attached disks. However, using local disks typically provides better performance than remotely attached disks, such as SSD Instance Store Volumes instead of EBS Volumes on AWS or Local SSDs instead of Persistent Disks on GCE. As of Kubernetes v1.14, [`local` volumes](https://kubernetes.io/docs/concepts/storage/volumes/#local) can be used.
```
Comment (MichaelTrestman) [inline v21.2/kubernetes-performance.md:161]:
```suggestion
Up to this point, we have assumed the use of auto-provisioned, remotely attached disks. However, local disks typically provide better performance than remotely attached disks. For example, SSD Instance Store Volumes outperform EBS Volumes on AWS, and Local SSDs outperform Persistent Disks on GCE. As of v1.14, Kubernetes supports [`local` volumes](https://kubernetes.io/docs/concepts/storage/volumes/#local).
```

**Review by taroface — COMMENTED** [review]
TFTR! This doc is very old and in need of broader updates. I agree with all your copy improvements.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/14340#-:-N6FInfv5UFQyyMsoe0N:b-kxz8pq)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @MichaelTrestman)

___
*[`v21.2/kubernetes-performance.md` line 23 at r1](https://reviewable.io/reviews/cockroachdb/docs/14340#-N5CKafSK9jQxmoXogYi-r1-23:-N6FI7ry4JKK6AbfA5eH:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/c28ed43e361ae490d2f9326cbd817f1d466d9aaf/v21.2/kubernetes-performance.md#L23)):*
<details><summary><i>Previously, MichaelTrestman (Michael Trestman) wrote…</i></summary><blockquote>

```suggestion
The following sections show how to modify excerpts from our provided Kubernetes configuration YAML files. You can find the most up-to-date versions of these files on Github, [one for running CockroachDB in secure mode](https://github.com/cockroachdb/cockroach/blob/master/cloud/kubernetes/cockroachdb-statefulset-secure.yaml) and one for [running CockroachDB in insecure mode](https://github.com/cockroachdb/cockroach/blob/master/cloud/kubernetes/cockroachdb-statefulset.yaml).
```
</blockquote></details>

Done.
___
*[`v21.2/kubernetes-performance.md` line 161 at r1](https://reviewable.io/reviews/cockroachdb/docs/14340#-N5CKafSK9jQxmoXogYi-r1-161:-N6FIEgr8eCcnvwl_kR2:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/c28ed43e361ae490d2f9326cbd817f1d466d9aaf/v21.2/kubernetes-performance.md#L161)):*
<details><summary><i>Previously, MichaelTrestman (Michael Trestman) wrote…</i></summary><blockquote>

```suggestion
Up to this point, we have assumed the use of auto-provisioned, remotely attached disks. However, local disks typically provide better performance than remotely attached disks. For example, SSD Instance Store Volumes outperform EBS Volumes on AWS, and Local SSDs outperform Persistent Disks on GCE. As of v1.14, Kubernetes supports [`local` volumes](https://kubernetes.io/docs/concepts/storage/volumes/#local).
```
</blockquote></details>

Done.
___
*[`v21.2/kubernetes-performance.md` line 409 at r1](https://reviewable.io/reviews/cockroachdb/docs/14340#-N5CKafSK9jQxmoXogYi-r1-409:-N6FIQHqCe6t8iGOwEpC:bwawn8j) ([raw file](https://github.com/cockroachdb/docs/blob/c28ed43e361ae490d2f9326cbd817f1d466d9aaf/v21.2/kubernetes-performance.md#L409)):*
<details><summary><i>Previously, MichaelTrestman (Michael Trestman) wrote…</i></summary><blockquote>

"We strongly recommend **against** using DaemonSets due to their weaker fault tolerance and data survivability capabilities, in comparison to <fill in the blank>."
I think it's good to clearly specify the contrast class whenever making a comparison like that. DaemonSet are weaker and less fault tolerant than what... Deployments? Regular style?
</blockquote></details>

StatefulSets! Made the clarification.
___
*[`v22.1/kubernetes-performance.md` line 161 at r1](https://reviewable.io/reviews/cockroachdb/docs/14340#-N6EtkE7BV3IKakTZH85:-N6FImNMEKqHVihTEzGA:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/c28ed43e361ae490d2f9326cbd817f1d466d9aaf/v22.1/kubernetes-performance.md#L161)):*
<details><summary><i>Previously, MichaelTrestman (Michael Trestman) wrote…</i></summary><blockquote>

If you like my suggestions on this copy on the other page, you could apply it here as well.
</blockquote></details>

Done.

**Review by MichaelTrestman — APPROVED** [review]
> Reviewed 2 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/14340)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @taroface)

**Pull request merged**